UnitTests/UnitTests.UWP/UI/Controls/Test_RichSuggestBox.cs

```diff
@@ -0,0 +1,272 @@
+                Assert.AreEqual(FormatEffect.On, range.CharacterFormat.Italic, "Token is expected to be italic.");
+                Assert.AreEqual(9, range.CharacterFormat.Size, "Unexpected token font size.");
+            });
+        }
+
+        [TestCategory(nameof(RichSuggestBox))]
+        [TestMethod]
+        [DataRow("@Token1", "@Token2")]
+        [DataRow("@Token1", "#Token2")]
+        [DataRow("#Token1", "@Token2")]
+        public async Task Test_RichSuggestBox_DeleteTokens(string token1, string token2)
+        {
+            await App.DispatcherQueue.EnqueueAsync(async () =>
+            {
+                var rsb = new RichSuggestBox() { Prefixes = "@#" };
+                await SetTestContentAsync(rsb);
+                var document = rsb.TextDocument;
+                var selection = document.Selection;
+
+                await AddTokenAsync(rsb, token1);
+                await AddTokenAsync(rsb, token2);
+
+                Assert.AreEqual(2, rsb.Tokens.Count, "Unexpected token count after adding.");
+
+                // Delete token as a whole
+                selection.Delete(TextRangeUnit.Character, -1);
+                selection.Delete(TextRangeUnit.Link, -1);
+                await Task.Delay(10);
+
+                Assert.AreEqual(1, rsb.Tokens.Count, "Unexpected token count after deleting token 2");
+
+                // Partially delete a token
+                selection.Delete(TextRangeUnit.Character, -2);
+                await Task.Delay(10);
+
+                Assert.AreEqual(0, rsb.Tokens.Count, "Unexpected token count after deleting token 1");
+            });
+        }
+
+        [TestCategory(nameof(RichSuggestBox))]
+        [TestMethod]
+        public async Task Test_RichSuggestBox_ReplaceToken()
+        {
+            await App.DispatcherQueue.EnqueueAsync(async () =>
+            {
+                var rsb = new RichSuggestBox() { Prefixes = "@" };
+                await SetTestContentAsync(rsb);
+                var document = rsb.TextDocument;
+                var selection = document.Selection;
+
+                await AddTokenAsync(rsb, "@Before");
+                var tokenBefore = rsb.Tokens[0];
+                AssertToken(rsb, tokenBefore, "@Before");
+
+                selection.Delete(TextRangeUnit.Character, -2);
+                await Task.Delay(10);
+
+                await AddTokenAsync(rsb, "@After");
+                var tokenAfter = rsb.Tokens[0];
+                AssertToken(rsb, tokenAfter, "@After");
+
+                Assert.AreNotSame(tokenBefore, tokenAfter, "Token before and token after are the same.");
+                Assert.AreNotEqual(tokenBefore.Id, tokenAfter.Id, "Token ID before and token ID after are the same.");
+            });
+        }
+
+        [TestCategory(nameof(RichSuggestBox))]
+        [TestMethod]
+        public async Task Test_RichSuggestBox_FormatReset()
```

Comment:
We could also try a single test in here with just basically constructing the control and checking if it's not null for instance. At least to see if that works.